java/connector-node/risingwave-sink-dynamic/src/main/java/com/risingwave/connector/FlinkDynamicAdaptFactory.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright 2023 RisingWave Labs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.risingwave.connector;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.risingwave.connector.api.TableSchema;
+import com.risingwave.connector.api.sink.SinkFactory;
+import com.risingwave.connector.api.sink.SinkWriter;
+import com.risingwave.connector.api.sink.SinkWriterV1;
+import com.risingwave.proto.Catalog;
+import java.util.*;
+import org.apache.flink.api.common.io.OutputFormat;
+import org.apache.flink.api.connector.sink.Sink;
+import org.apache.flink.configuration.ConfigOption;
+import org.apache.flink.configuration.Configuration;
+import org.apache.flink.streaming.api.functions.sink.OutputFormatSinkFunction;
+import org.apache.flink.streaming.api.functions.sink.SinkFunction;
+import org.apache.flink.table.api.Schema;
+import org.apache.flink.table.api.TableException;
+import org.apache.flink.table.catalog.*;
+import org.apache.flink.table.connector.sink.*;
+import org.apache.flink.table.data.RowData;
+import org.apache.flink.table.factories.DynamicTableSinkFactory;
+import org.apache.flink.table.factories.FactoryUtil;
+import org.apache.flink.table.runtime.connector.sink.SinkRuntimeProviderContext;
+
+public class FlinkDynamicAdaptFactory implements SinkFactory {
+
+    public DynamicTableSink.SinkRuntimeProvider buildDynamicTableSinkProvider(
+            TableSchema tableSchema, FlinkDynamicAdaptConfig config) {
+        List<Column> flinkColumns = FlinkDynamicAdaptUtil.getFlinkColumnsFromSchema(tableSchema);
+
+        // Start with the default value, and add as needed later
+        ObjectIdentifier objectIdentifier =
+                ObjectIdentifier.of("catalog_name", "database_name", "obj_name");
+        ClassLoader contextClassLoader = Thread.currentThread().getContextClassLoader();
+        DynamicTableSinkFactory dynamicTableSinkFactory =
```

Comment:
I think we don't have to leverage flink's utils to build the factory. Currently we can manually create the factory by the connector name (e.g. create the http dynamic table sink factory when we check the connector name is `http`).

Reply:
Using FlinkUtil, when adding a new sink, we can allow the class loader to load it directly, without the need to add new factory logic.

Reply:
The class loader in flink is to support loading customized sinks implemented by users, which is unnecessary for us because we have a whitelist for supported sinks. Besides, the class loader logic is too implicit, and we cannot ensure that the class loader is loading exactly the sink factory we expect. Instead, implementing a simple wrapper factory for each flink sink is acceptable, and will make the loaded sink factory more explicit.

 Moreover, using the flink class loader will force us to use the same connector name as the name defined in the flink jar, and we cannot do any rename on it (e.g. http-sink -> flink-http-sink).